### PR TITLE
Feature/do 1394 use secrets manager for integrated api key

### PR DESCRIPTION
**Pull request by crispy101**
**Description of the proposed changes**  

* Introduce SecretsManager-based authentication parameter: tokenSecret - this is used for token-URL validation for both Prerender and Recache services
* Deprecate insecure parameters: tokenList and tokenUrlAssociation
* Improve code and instruction readability

**Screenshots (if applicable)**  

* 

**Other solutions considered (if any)**  

* 

**Notes to reviewers**  

🛈  When you've finished leaving feedback, please add a final comment to the PR tagging the author, letting them know that you have finished leaving feedback

### CODE REVIEW

**Review by TheOrangePuff — CHANGES_REQUESTED** [review]
A few comments :slightly_smiling_face: 

Have these files had prettier run over them? Some of the formatting looks a bit off :thinking:

File: packages/prerender-fargate/README.md [hunk 1]
```diff
@@ -22,67 +26,75 @@ To use the PrerenderFargate construct, you can instantiate it with suitable Prer
 
 ### `domainName` (string)
 
-- The domain name to prerender.
+- The domain name to Prerender service.
+
+### `tokenSecret` (strings)
+
+- A pre-configured AWS SecretsManager Secret name with the value being in the format of `Map<String: String[]>`, for example, `{"token1": "https://www.example1.com,https://www.mydomain1.com", "token2":"https://www.example2.com,https://www.mydomain2.com"}`. This map is referenced for Prerender and Recaching service authentication.
```
[the remark this answers is not in the document]
Comment (crispy101) [reply packages/prerender-fargate/README.md:33]: Good idea! Added a screenshot.

File: packages/prerender-fargate/README.md [hunk 2]
```diff
@@ -22,67 +26,75 @@ To use the PrerenderFargate construct, you can instantiate it with suitable Prer
...
+- Whether to enable caching of HTTP redirects. Default: false
 
 ### `enableS3Endpoint` (boolean, optional)
 
-- Whether to enable the [VPC endpoint](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html) for S3.
-
-### `tokenUrlAssociation` (PrerenderTokenUrlAssociationOptions, optional)
-
-- Configuration for associating tokens with specific domain URLs. During the recaching process, these tokens will be used to validate the request.
+- Whether to enable the [VPC endpoint](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html) for S3. Default: "false"
```
[the remark this answers is not in the document]
Comment (crispy101) [reply packages/prerender-fargate/README.md]: Nice finding. Done.

**Review by crispy101 — COMMENTED** [review]
Thanks @TheOrangePuff for the inputs. I've made some changes.

File: packages/prerender-fargate/lib/prerender/server.js [hunk 3]
```diff
@@ -120,39 +127,47 @@ if (process.env.ENABLE_REDIRECT_CACHE.toLowerCase() === 'true'){
...
-            s3.putObject({
-                Key: key,
-                ContentType: 'text/html;charset=UTF-8',
-                StorageClass: 'REDUCED_REDUNDANCY',
-                Body: req.prerender.content,
-                Metadata: s3Metadata
-            }, function(err, result) {
-                console.log(result);
-                if (err) console.error(err);
+                    s3.getObject({
```
[the remark this answers is not in the document]
Comment (TheOrangePuff) [reply packages/prerender-fargate/lib/prerender/server.js:145]: Repeated logic isn't the only reason to pull something out into a function. 

Having so much nesting makes it hard to read. A cacheObject function makes it clear that you're getting the file from s3 and then caching it.

[the remark this answers is not in the document]
Comment (crispy101) [reply packages/prerender-fargate/lib/prerender/server.js:164]: Roger that. Updated.

**Review by TheOrangePuff — APPROVED** [review]

**Pull request merged**